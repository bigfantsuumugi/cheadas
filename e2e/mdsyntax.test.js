const path = require('path')
const {
  launchDevServer,
  launchBrowser,
  prepareFixture,
  getEl
} = require('./common/utils')

const binPath = path.resolve(__dirname, '../bin/vitepress.js')
const fixtureDir = path.join(__dirname, '../playground/mdsyntax')
const tempDir = path.join(__dirname, '../.fixtures/mdsyntax')

let devServer, browser

beforeAll(async () => {
  await prepareFixture(fixtureDir, tempDir)
  devServer = await launchDevServer(binPath, tempDir)
  browser = await launchBrowser()
})

afterAll(async () => {
  if (browser) await browser.close()
  if (devServer) await devServer.close()
})

describe('markdown syntax', () => {
  test('basic', async () => {
    const page = await browser.newPage()
    page.setViewport({ width: 1600, height: 900 })

    await page.goto('http://localhost:3000/basic.html')
    await page.waitFor('main > div.content')

    const el = await getEl(page, 'main > div.content')
    const html = await el.evaluate((e) => e.outerHTML)

    await el.screenshot({ path: path.join(tempDir, 'basic.png') })
    expect(html).toMatchSnapshot()
  })
})
