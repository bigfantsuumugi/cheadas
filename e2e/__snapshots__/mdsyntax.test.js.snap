// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown syntax basic 1`] = `
"<div class=\\"content\\"><div><!-- https://www.markdownguide.org/basic-syntax --><h1 id=\\"basic-syntax\\"><a class=\\"header-anchor\\" href=\\"#basic-syntax\\" aria-hidden=\\"true\\">#</a> Basic Syntax</h1><h2 id=\\"headings\\"><a class=\\"header-anchor\\" href=\\"#headings\\" aria-hidden=\\"true\\">#</a> Headings</h2><p>To create a heading, add number signs (#) in front of a word or phrase. The number of number signs you use should correspond to the heading level. For example, to create a heading level three (<code>&lt;h3&gt;</code>), use three number signs (e.g., ### My Header).</p><h1 id=\\"heading-level-1\\"><a class=\\"header-anchor\\" href=\\"#heading-level-1\\" aria-hidden=\\"true\\">#</a> Heading level 1</h1><h2 id=\\"heading-level-2\\"><a class=\\"header-anchor\\" href=\\"#heading-level-2\\" aria-hidden=\\"true\\">#</a> Heading level 2</h2><h3 id=\\"heading-level-3\\"><a class=\\"header-anchor\\" href=\\"#heading-level-3\\" aria-hidden=\\"true\\">#</a> Heading level 3</h3><h4 id=\\"heading-level-4\\"><a class=\\"header-anchor\\" href=\\"#heading-level-4\\" aria-hidden=\\"true\\">#</a> Heading level 4</h4><h5 id=\\"heading-level-5\\"><a class=\\"header-anchor\\" href=\\"#heading-level-5\\" aria-hidden=\\"true\\">#</a> Heading level 5</h5><h6 id=\\"heading-level-6\\"><a class=\\"header-anchor\\" href=\\"#heading-level-6\\" aria-hidden=\\"true\\">#</a> Heading level 6</h6><ul><li>Alternate Syntax:</li></ul><h1 id=\\"heading-level-1-2\\"><a class=\\"header-anchor\\" href=\\"#heading-level-1-2\\" aria-hidden=\\"true\\">#</a> Heading level 1</h1><h2 id=\\"heading-level-2-2\\"><a class=\\"header-anchor\\" href=\\"#heading-level-2-2\\" aria-hidden=\\"true\\">#</a> Heading level 2</h2><h2 id=\\"paragraphs\\"><a class=\\"header-anchor\\" href=\\"#paragraphs\\" aria-hidden=\\"true\\">#</a> Paragraphs</h2><p>I really like using Markdown.</p><p>I think I'll use it to format all of my documents from now on.</p><h2 id=\\"line-breaks\\"><a class=\\"header-anchor\\" href=\\"#line-breaks\\" aria-hidden=\\"true\\">#</a> Line Breaks</h2><p>This is the first line.<br> And this is the second line.</p><p>First line with the HTML tag after.<br> And the next line.</p><h2 id=\\"emphasis\\"><a class=\\"header-anchor\\" href=\\"#emphasis\\" aria-hidden=\\"true\\">#</a> Emphasis</h2><p>I just love <strong>bold text</strong>.</p><p>Love<strong>is</strong>bold</p><p>Italicized text is the <em>cat's meow</em>.</p><p>Italicized text is the <em>cat's meow</em>.</p><p>A<em>cat</em>meow</p><p>This text is <em><strong>really important</strong></em>.</p><p>This text is <em><strong>really important</strong></em>.</p><p>This text is <strong><em>really important</em></strong>.</p><p>This text is <strong><em>really important</em></strong>.</p><p>This is really<em><strong>very</strong></em>important text.</p><h2 id=\\"blockquotes\\"><a class=\\"header-anchor\\" href=\\"#blockquotes\\" aria-hidden=\\"true\\">#</a> Blockquotes</h2><blockquote><p>Dorothy followed her through many of the beautiful rooms in her castle.</p></blockquote><ul><li>Blockquotes with Multiple Paragraphs:</li></ul><blockquote><p>Dorothy followed her through many of the beautiful rooms in her castle.</p><p>The Witch bade her clean the pots and kettles and sweep the floor and keep the fire fed with wood.</p></blockquote><ul><li>Nested Blockquotes:</li></ul><blockquote><p>Dorothy followed her through many of the beautiful rooms in her castle.</p><blockquote><p>The Witch bade her clean the pots and kettles and sweep the floor and keep the fire fed with wood.</p></blockquote></blockquote><ul><li>Blockquotes with Other Elements:</li></ul><blockquote><h4 id=\\"the-quarterly-results-look-great\\"><a class=\\"header-anchor\\" href=\\"#the-quarterly-results-look-great\\" aria-hidden=\\"true\\">#</a> The quarterly results look great!</h4><ul><li>Revenue was off the chart.</li><li>Profits were higher than ever.</li></ul><p><em>Everything</em> is going according to <strong>plan</strong>.</p></blockquote><h2 id=\\"lists\\"><a class=\\"header-anchor\\" href=\\"#lists\\" aria-hidden=\\"true\\">#</a> Lists</h2><ol><li><p>First item</p></li><li><p>Second item</p></li><li><p>Third item</p></li><li><p>Fourth item</p></li><li><p>First item</p></li><li><p>Second item</p></li><li><p>Third item</p></li><li><p>Fourth item</p></li><li><p>First item</p></li><li><p>Second item</p></li><li><p>Third item</p></li><li><p>Fourth item</p></li><li><p>First item</p></li><li><p>Second item</p></li><li><p>Third item</p><ol><li>Indented item</li><li>Indented item</li></ol></li><li><p>Fourth item</p></li></ol><ul><li>First item</li><li>Second item</li><li>Third item</li><li>Fourth item</li></ul><ul><li>First item</li><li>Second item</li><li>Third item</li><li>Fourth item</li></ul><ul><li>First item</li><li>Second item</li><li>Third item</li><li>Fourth item</li></ul><ul><li>First item</li><li>Second item</li><li>Third item <ul><li>Indented item</li><li>Indented item</li></ul></li><li>Fourth item</li></ul><ul><li><p>Adding Elements in Lists</p></li><li><p>This is the first list item.</p></li><li><p>Here's the second list item.</p><p>I need to add another paragraph below the second list item.</p></li><li><p>And here's the third list item.</p></li><li><p>This is the first list item.</p></li><li><p>Here's the second list item.</p><blockquote><p>A blockquote would look great below the second list item.</p></blockquote></li><li><p>And here's the third list item.</p></li></ul><ol><li><p>Open the file.</p></li><li><p>Find the following code block on line 21:</p><pre><code>&lt;html&gt;
  &lt;head&gt;
    &lt;title&gt;Test&lt;/title&gt;
  &lt;/head&gt;
</code></pre></li><li><p>Update the title to match the name of your website.</p></li><li><p>Open the file containing the Linux mascot.</p></li><li><p>Marvel at its beauty.</p><p><img src=\\"/assets/images/tux.png\\" alt=\\"Tux, the Linux mascot\\"></p></li><li><p>Close the file.</p></li><li><p>First item</p></li><li><p>Second item</p></li><li><p>Third item</p><ul><li>Indented item</li><li>Indented item</li></ul></li><li><p>Fourth item</p></li></ol><h2 id=\\"code\\"><a class=\\"header-anchor\\" href=\\"#code\\" aria-hidden=\\"true\\">#</a> Code</h2><p>At the command prompt, type <code>nano</code>.</p><p><code>Use \`code\` in your Markdown file.</code></p><h2 id=\\"horizontal-rules\\"><a class=\\"header-anchor\\" href=\\"#horizontal-rules\\" aria-hidden=\\"true\\">#</a> Horizontal Rules</h2><p>Try to put a blank line before...</p><hr><p>...and after a horizontal rule.</p><h2 id=\\"links\\"><a class=\\"header-anchor\\" href=\\"#links\\" aria-hidden=\\"true\\">#</a> Links</h2><p>My favorite search engine is <a href=\\"https://duckduckgo.com\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Duck Duck Go</a>.</p><p>My favorite search engine is <a href=\\"https://duckduckgo.com\\" title=\\"The best search engine for privacy\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Duck Duck Go</a>.</p><p><a href=\\"https://www.markdownguide.org\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">https://www.markdownguide.org</a><a href=\\"./mailto:fake@example.com.html\\">fake@example.com</a></p><h2 id=\\"images\\"><a class=\\"header-anchor\\" href=\\"#images\\" aria-hidden=\\"true\\">#</a> Images</h2><p><img src=\\"/assets/images/philly-magic-gardens.jpg\\" alt=\\"Philadelphia's Magic Gardens. This place was so cool!\\" title=\\"Philadelphia's Magic Gardens\\"></p><h2 id=\\"escaping-characters\\"><a class=\\"header-anchor\\" href=\\"#escaping-characters\\" aria-hidden=\\"true\\">#</a> Escaping Characters</h2><p>* Without the backslash, this would be a bullet in an unordered list. \\\\ \` { } [ ] ...</p><h2 id=\\"html\\"><a class=\\"header-anchor\\" href=\\"#html\\" aria-hidden=\\"true\\">#</a> HTML</h2><p>This <strong>word</strong> is bold. This <em>word</em> is italic.</p></div><div class=\\"links-wrapper\\"><div class=\\"prev-link\\"><!--v-if--></div><div class=\\"next-link\\"><!--v-if--></div></div></div>"
`;
